### PR TITLE
Add `getCurrentChainIO`, use it for `BlockContext` acquisition

### PR DESCRIPTION
**Pull request by amesgen**
Motivation: see if this has any impact on performance, cf #461

### CODE REVIEW

File: ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/Query.hs [hunk 1]
```diff
@@ -60,6 +62,20 @@ import           Ouroboros.Network.Block (MaxSlotNo, maxSlotNoFromWithOrigin)
 -- In the latter case, we don't take blocks already in the ImmutableDB into
 -- account, as we know they /must/ have been \"immutable\" at some point, and,
 -- therefore, /must/ still be \"immutable\".
+getCurrentChain'
```
Comment (nfrisby) [inline ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/Query.hs:65]: Minor: If you defined `getCurrentChain'` last, then would the diff be shorter/better aligned?

**Review by nfrisby — COMMENTED** [review]
I agree that this diff does what we had discussed on the phone call today re Issue 461.

File: ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/Query.hs (hunk 1 above)
Comment (amesgen) [reply ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/Query.hs:65]: Right, that seems to be the case, will change :+1:

**Comment by amesgen** [comment]
This is no longer relevant, see #513

**Pull request closed without merging**